lib/Conversion/TritonGPUToLLVM/TypeConverter.cpp

```diff
@@ -79,6 +79,35 @@ Value TritonGPUToLLVMTypeConverter::packLLElements(
   return llvmStruct;
 }
 
+SmallVector<Value> TritonGPUToLLVMTypeConverter::packMfmaOperand(
+    const SmallVector<Value> &inValues, Type srcTy,
+    ConversionPatternRewriter &rewriter, Location loc) {
+  auto tensorTy = srcTy.dyn_cast<RankedTensorType>();
+  if (!tensorTy)
+    return inValues;
+  auto encoding = tensorTy.getEncoding().dyn_cast<DotOperandEncodingAttr>();
+  if (!(encoding && encoding.getParent().isa<MfmaEncodingAttr>())) {
+    return inValues;
+  }
+
+  SmallVector<Value> result;
+  auto structType = this->convertType(srcTy).dyn_cast<LLVM::LLVMStructType>();
+  auto elementTypes = structType.getBody();
+  assert(elementTypes.size() > 0);
+  auto vecTy = elementTypes[0];
+  unsigned size = 4;
```

Comment:
is this hard-coded due to the assumption of mfma_32x32x8f16?

Reply:
Change to call `unsigned size = vecTy.getNumElements();` for the vec size.